dask/array/tests/test_cupy_percentile.py

```diff
@@ -1,35 +1,44 @@
 import numpy as np
 import pytest
+from packaging.version import parse as parse_version
 
 pytestmark = pytest.mark.gpu
 
 import dask.array as da
-from dask.array.numpy_compat import _numpy_120
+from dask.array.numpy_compat import _numpy_120, _numpy_122
 from dask.array.utils import assert_eq, same_keys
 
 cupy = pytest.importorskip("cupy")
 
+cupy_version = parse_version(cupy.__version__)
+_numpy_gt_122_cupy_lt_101 = _numpy_122 & (cupy_version < parse_version("10.1"))
+
 
 @pytest.mark.skipif(not _numpy_120, reason="NEP-35 is not available")
 def test_percentile():
     d = da.from_array(cupy.ones((16,)), chunks=(4,))
     qs = np.array([0, 50, 100])
 
-    assert_eq(
-        da.percentile(d, qs, interpolation="midpoint"),
-        np.array([1, 1, 1], dtype=d.dtype),
-        check_type=False,
-    )
+    if _numpy_gt_122_cupy_lt_101:
+        with pytest.raises(FutureWarning):
+            result = da.percentile(d, qs, interpolation="midpoint")
+    else:
+        result = da.percentile(d, qs, method="midpoint")
```

Comment:
@jsignell thanks for working on the CuPy changes. I think we still need a conditional for when both NumPy < 1.22 and CuPy < 10.1. How about the following:

```python
if _cupy_lt_101:
    ctx = pytest.raises(FutureWarning) if _numpy_122 else contextlib.nullcontext()
    with ctx:
        result = da.percentile(d, qs, interpolation="midpoint")
else:
    result = da.percentile(d, qs, method="midpoint")
```

The equivalent would apply for tests below.

Reply:
The FutureWarning comes from dask array regardless of numpy version. After looking at this again, I'm thinking that we can probably just replace `intepolation` with `method` in the tests and things will just work. 

```suggestion
    result = da.percentile(d, qs, method="midpoint")
```

Reply:
Ah yes, I wasn't attentive to the fact that you handle that with the internal method and it dispatches with the correct argument name depending on the NumPy version. However, this may be problematic for use new Dask with old NumPy with `__array_function__`, since it depends on the exact argument name. Therefore, if anybody uses NumPy 1.21 or older and tries to `np.percentile(dask_a, dask_q, interpolation="midpoint")` it won't work anymore. What do you think about keeping `interpolation` and raise a Dask `FutureWarning` just like NumPy is doing until Dask drops support for NumPy < 1.22?

Reply:
I think that is the current behavior becuase I added a numpy_compat layer where I check the numpy version and then do method or interpolation depending on the numpy version. Let me comment on that bit.